safehttp/incoming_request.go

```diff
@@ -14,9 +14,66 @@
 
 package safehttp
 
-import "net/http"
+import (
+	"errors"
+	"net/http"
+	"strings"
+)
 
 // IncomingRequest TODO
 type IncomingRequest struct {
 	req *http.Request
 }
+
+// QueryForm TODO
+func (r IncomingRequest) QueryForm() (*Form, error) {
```

Comment:
maybe we should wrap this and the other getters in a `sync.Once` call or have another memoization for the values, otherwise every time the getter is called a lot of work needs to be done.

Reply:
Oh, true, thanks for this! Resolved in [d511d47](url)